fluffy/slurm_api.py

```diff
@@ -9,14 +9,22 @@
 
 
 class SlurmAPI:
-    """A API to SLURM"""
+    """An API to SLURM"""
 
-    def __init__(self, account: str, time: str, out_dir: Path, partition: str = None):
+    def __init__(self,slurm_settings: dict, out_dir: Path):
```

Comment:
Why did you hide all variables in a dictionary here?

Reply:
The slurm settings are retrieved from the config file (the slurm dictionary of the json config file), I realized this is nesecary after testing fluffy on hasta:

For instance, on Hasta you will get a sbatch error if you ask for a core or node partition.
On hasta, you need to use the ntasks variable to select the number of cores.

In short the user will have to decide upon their own SLURM variables to use, by editing the config file

Reply:
Yes I realise that but would it not be better to set them explicitly? It's not that many options.

Reply:
Maybe we could add some kind of check when reading the config file (as we do for the references etc)?
I agree it is nice to be able to check if the settings make sense or not.

We can assume some settings, such as account, and time. And it seems reasonable to either use partition to ask for cores/node, or ntasks to specify number of cores.

Overall, there are nearly 100 variables available (although some of these seem a bit obscure), and we have no idea on how the users have configured their cluster.

Reply:
In reality which ones will we use? In general I think it is better to solve the problems if they arise, at the moment we will only use a few so it is more transparent if they are passed as arguments and not just a dictionary.